### PR TITLE
First part of Venue and Organizer Insertion support

### PR DESCRIPTION
**Pull request by lucatume**
Tickets: https://central.tri.be/issues/79737 and  https://central.tri.be/issues/79740

Builds on https://github.com/moderntribe/tribe-common/pull/416

This PR adds support to the REST API to Insert (POST) Venues and Organizers (with meta and image) plus tests.
What is missing is the mechanic that will prevent duplication of venues and organizers when inserting theme by data (e.g. name) and not ID (the issue we discusses in Code Review); this is the first part of the work to make sure we cover the "dumb" before we try to cover the "smart".

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Space found before comma in function call | [Post_Repository.php:177](https://github.com/moderntribe/the-events-calendar/blob/d84340c15b91756d3c12271848b3d0598f554b76/src/Tribe/REST/V1/Post_Repository.php#L177)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Space found before comma in function call | [Post_Repository.php:177](https://github.com/moderntribe/the-events-calendar/blob/97e4adf4a0343417750e9f4d6fe2a1525a87c9c0/src/Tribe/REST/V1/Post_Repository.php#L177)

> via jenkins/codesniffer-phpcs

**Comment by lucatume** [comment]
This is ready for review.

File: src/Tribe/Organizer.php [hunk 1]
```diff
@@ -305,15 +329,39 @@ public function create( $data, $post_status = 'publish' ) {
...
 
+			$data = new Tribe__Data( $data, false );
+
 			$postdata = array(
 				'post_title'   => $title,
 				'post_content' => $content,
 				'post_name'    => $slug,
 				'post_type'    => self::POSTTYPE,
-				'post_status'  => $post_status,
+				'post_status'   => isset( $data['post_status'] ) ? $data['post_status'] : $post_status,
```
Comment (borkweb) [inline src/Tribe/Organizer.php]: @bordoni would probably suggest you do this instead:

```php
'post_status' => Tribe__Utils__Array::get( $data, 'post_status', $post_status ),
```

File: src/Tribe/Venue.php [hunk 2]
```diff
@@ -306,29 +313,79 @@ public function save_meta( $venue_id, $data ) {
...
-				'post_title'  => $title,
-				'post_content' => $content,
-				'post_name'   => $slug,
-				'post_type'   => self::POSTTYPE,
-				'post_status' => $post_status,
+				'post_title'    => $title,
+				'post_content'  => $content,
+				'post_name'     => $slug,
+				'post_type'     => self::POSTTYPE,
+				'post_status'   => isset( $data['post_status'] ) ? $data['post_status'] : $post_status,
```
Comment (borkweb) [inline src/Tribe/Venue.php]: `Tribe__Utils__Array::get` here as well.

**Comment by lucatume** [comment]
Utils everywhere, fixed.

**Pull request merged**